src/core/fair_queue.cc

```diff
@@ -290,7 +298,7 @@ void fair_queue::register_priority_class(class_id id, uint32_t shares) {
 
 void fair_queue::unregister_priority_class(class_id id) {
     auto& pclass = _priority_classes[id];
-    assert(pclass && pclass->_queue.empty());
+    assert(pclass && !pclass->has_noncancelled_entry());
```

Comment:
I think that this line deserves a comment summing up recent discussion here -- despite it's technically OK to destruct the class with queued requests, we still want to make sure that the rest of the code picked up the requests it queued previously

Reply:
Added a comment.